### PR TITLE
Implement VMMetrics factory and config V2

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Github issue: https://github.com/open-telemetry/opentelemetry-service/issues/35

Testing done: make && make otelsvc

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
>LGTM. Do you want to clean up the old config? 

@songy23 yes, it is in my todo list. I just want to first have feature parity in new config implementation and then start deleting old config support.

**Comment by tigrannajaryan** [comment]
Keeping the support for old config allows to build old occollector executable and run it and compare with otelsvc behavior. Very useful for testing.

File: receiver/vmmetricsreceiver/metrics_receiver.go [hunk 1]
```diff
@@ -76,8 +76,15 @@ func New(v *viper.Viper, consumer consumer.MetricsConsumer) (*Receiver, error) {
...
+const metricsSource string = "VMMetrics"
+
+// MetricsSource returns the name of the metrics data source.
+func (vmr *Receiver) MetricsSource() string {
+	return metricsSource
+}
+
 // StartMetricsReception scrapes VM metrics based on the OS platform.
-func (vmr *Receiver) StartMetricsReception(ctx context.Context, asyncErrorChan chan<- error) error {
+func (vmr *Receiver) StartMetricsReception(host receiver.Host) error {
```
Comment (pjanotti) [inline receiver/vmmetricsreceiver/metrics_receiver.go]: 👍

**Comment by tigrannajaryan** [comment]
Results as seen in Prometheus:

image

**Pull request merged**